release/create-release-candidate-branch.sh

```diff
@@ -150,22 +193,28 @@ update_code() {
 	fi
 
 	# Update operator version for the integration tests
-	# this is used when installing the operators.
+	# (used when installing the operators).
 	yq -i ".releases.tests.products[].operatorVersion |= sub(\"0.0.0-dev\", \"${RELEASE_TAG}\")" "$1/tests/release.yaml"
 
-	# Some tests perform label inspection and for these cases only specific labels should be updated.
+	# do this for patch releases/release candidates too.
+	# i.e. replace 24.11.0-rc1 with 24.11.0, 24.7.0 with 24.7.1 etc.
+	yq -i "(.releases.tests.products[].operatorVersion | select(. == \"${RELEASE}*\")) |= \"${RELEASE_TAG}\"" "$1/tests/release.yaml"
+
+	# Some tests perform **label** inspection and for (only) these cases specific labels should be updated.
 	# N.B. don't do this for all test files as not all images will necessarily exist for the given release tag.
-	find "$1/tests/templates/kuttl" -type f -print0 | xargs -0 sed -i "/app.kubernetes.io\/version/{ s/stackable0.0.0-dev/stackable$RELEASE_TAG/ }"
+	find "$1/tests/templates/kuttl" -type f -print0 | xargs -0 sed -E -i "s#(app\.kubernetes\.io/version: \".*-stackable)[^\"]*#\1$RELEASE_TAG#"
 }
 
 push_branch() {
 	if $PUSH; then
 		echo "Pushing changes..."
-		git push "${REPOSITORY}" "${RELEASE_BRANCH}"
-		git push "${REPOSITORY}" "${RELEASE_TAG}"
-		git switch main
+		# the branch must be updated before the PR can be created
+		git push "${REPOSITORY}" "${PR_BRANCH}"
+		gh pr create --reviewer stackabletech/developers --base "${RELEASE_BRANCH}" --head "${PR_BRANCH}" --title "release: ${RELEASE_TAG}" --body "${PR_MSG}"
```

Comment:
suggestion: Use conventional-commits in PR titles.

```suggestion
		gh pr create --reviewer stackabletech/developers --base "${RELEASE_BRANCH}" --head "${PR_BRANCH}" --title "chore: Release ${RELEASE_TAG}" --body "${PR_MSG}"
```

Reply:
Another idea would be to construct these command lines once, and in the dry-run case, just append the `--dry-run` flag (assuming order isn't important).

Something like:

```shell
PUSH_CMD="git push '${REPOSITORY}' '${PR_BRANCH}'"
PR_CMD="gh pr create --reviewer stackabletech/developers --base '${RELEASE_BRANCH}' --head '${PR_BRANCH}' --title 'chore: Release ${RELEASE_TAG}' --body '${PR_MSG}'"

if $PUSH; then
  echo "Pushing changes..."
  bash -c "$PUSH_CMD"
  bash -c "$PR_CMD"
else
  echo "(Dry-run: not pushing...)"
  bash -c "$PUSH_CMD --dry-run"
  bash -c "$PR_CMD --dry-run"
fi
```

> [!NOTE]
> _I used a `bash` here so that shellcheck doesn't complain about executing the variable. Alternatively, you can disable shellcheck for these calls._